nexus/db-queries/src/db/queries/region_allocation.rs

```diff
@@ -117,10 +189,56 @@ pub fn allocation_query(
 
     let seed = seed.to_le_bytes().to_vec();
 
-    let size_delta =
-        params.block_size * params.blocks_per_extent * params.extent_count;
+    // Ensure that the multiplication doesn't overflow.
+    let requested_size: u64 = params
+        .block_size
+        .checked_mul(params.blocks_per_extent)
+        .ok_or(AllocationQueryError::RegionSizeOverflow)?
+        .checked_mul(params.extent_count)
+        .ok_or(AllocationQueryError::RegionSizeOverflow)?;
+
+    if requested_size > MAX_REGION_SIZE_BYTES {
+        return Err(AllocationQueryError::RequestedRegionOverMaxSize {
+            request: requested_size,
+            maximum: MAX_REGION_SIZE_BYTES,
+        });
+    }
+
+    // After the above check, unconditionally cast from u64 to i64. The value is
+    // low enough that this shouldn't truncate.
```

Comment:
FWIW I don't think it would truncate either way, I *think* it would fail with an error?

Reply:
after discussing it a bit, went with returning an Err instead of unwrapping in 7d9fed61d